clippy_lints/src/floating_point_arithmetic.rs

```diff
@@ -293,6 +293,92 @@ fn check_powf(cx: &LateContext<'_, '_>, expr: &Expr<'_>, args: &[Expr<'_>]) {
     }
 }
 
+fn check_powi(cx: &LateContext<'_, '_>, expr: &Expr<'_>, args: &[Expr<'_>]) {
+    // Check argument
+    if let Some((value, _)) = constant(cx, cx.tables, &args[1]) {
+        // TODO: need more specific check. this is too wide. remember also to include tests
+        if let Some(parent) = get_parent_expr(cx, expr) {
+            if let Some(grandparent) = get_parent_expr(cx, parent) {
+                if let ExprKind::MethodCall(PathSegment { ident: method_name, .. }, _, args) = grandparent.kind {
+                    if method_name.as_str() == "sqrt" && detect_hypot(cx, args).is_some() {
+                        return;
+                    }
+                }
+            }
+        }
+
+        let (lint, help, suggestion) = match value {
+            Int(2) => (
+                IMPRECISE_FLOPS,
+                "square can be computed more accurately",
```

Comment:
Is this really more accurate, or just more efficient?

Reply:
I don't have a definitive answer here. I have been assuming all these were precision lints just because the original issue talked about accuracy suggestions. I'm open to reclassifing this under `SUBOPTIMAL_FLOPS`. What do you think?

Reply:
Yeah I'd prefer that. They should generate the same answer, just one is more specialised and thus faster

Reply:
I'm trying to reclassify this one, but keep hitting into `failed to compile fixed code`.

The fix is generating code that gets linted by the `mul_add` lint for the `(x + y.powi(2)).sqrt()` case and I don't see how to allow mul_add while still fixing powi because now they're both under `suboptimal-flops`.

Reply:
Update: Managed to get past this one by changing powi to generate code that does not trigger mul_add.